src/lib/datepicker/datepicker.ts

```diff
@@ -302,10 +309,10 @@ export class MatDatepicker<D> implements OnDestroy, CanColor {
 
   /** Selects the given date */
   select(date: D): void {
-    let oldValue = this._selected;
-    this._selected = date;
-    if (!this._dateAdapter.sameDate(oldValue, this._selected)) {
-      this._selectedChanged.next(date);
+    let oldValue = this._selected.getFirstSelectedDate();
```

Comment:
Since this component knows its working with a single date selection you can use `asDate()` and `setSelection()` which feel a little more natural